File: frontend/src/libs/format-date.ts
const bulans = [
  "Januari",
  "Februari",
  "Maret",
  "April",
  "Mei",
  "Juni",
  "Juli",
  "Agustus",
  "September",
  "Oktober",
  "November",
  "Desember",
];

export default function (dateString: string, short: boolean = false) {
  const date = new Date(dateString);
  const bulan = bulans[date.getMonth()];

  let val = date.getDate().toString();
  val += " ";
  val += short ? bulan.slice(0, 3) : bulan;
  val += " ";
  val += date.getFullYear();

  return val;
}

File: mobile/lib/libs/format_date.dart
formatDate(String date, {bool short = false}) {
  final year = int.parse(date.substring(0, 4));
  final month = int.parse(date.substring(5, 7));
  final day = int.parse(date.substring(8, 10));

  final bulan = [
    'Januari',
    'Februari',
    'Maret',
    'April',
    'Mei',
    'Juni',
    'Juli',
    'Agustus',
    'September',
    'Oktober',
    'November',
    'Desember',
  ];

  return "${day.toString()} ${bulan[month].substring(0, short ? 3 : null)} ${year.toString()}";
}
